pipeline/notifications/check_message.py

```diff
@@ -0,0 +1,25 @@
+from typing import List
+
+import prefect
+from prefect import task
+
+from pipeline.types import Notification, User
+
+
+class Notifier:
+    def __init__(self):
+        pass
+
+    def check_msg_notifier(self, notif: Notification) -> List[User]:
+
+        # check if all users in match have a message
+        # if there are users who don't have a message, return a list of them for notification
+        users_with_msg = []
+        users_to_notify = []
+        for msg in notif.chatdata.messages:
```

Comment:
One big thing to think about (we should save this for a future PR): how will you deal with the timing of the pipeline runs?

For example, suppose user A messages the chat with user B and user B does not respond for the whole week. If we run our notification pipeline every night, we might send user B the same notification every day for the whole week.

Here are some ideas:
- A. Filter to only consider messages from the last 24 hours
- B. Save the timestamp of the last pipeline run and filter to only consider messages since then
- C. Store previously sent notifications and check against them to avoid creating a duplicate notification (or a notification of the same type to the same user)

Can you think of other approaches? What are the pros/cons of these approaches? Which would you choose to implement?

Reply:
What do you think about this @chunter3 ?

Reply:
Awesome! Great analysis. I agree on choosing option B.

If it's easier to build, we can start by implementing option A, with a variable number of hours, that way even if we run the pipeline weekly, we can adjust it.

Then, if we have time, we can refactor the pipeline to use option B.